test/Filesystem.test.cpp

```diff
@@ -28,6 +28,15 @@ TEST_F(FilesystemTest, dataPath) {
 	EXPECT_EQ("data/", fs.dataPath());
 }
 
+TEST_F(FilesystemTest, extension) {
+	EXPECT_EQ("txt", fs.extension("subdir/file.txt"));
+	EXPECT_EQ("txt", fs.extension("file.txt"));
+	EXPECT_EQ("reallyLongExtensionName", fs.extension("file.reallyLongExtensionName"));
+	EXPECT_EQ("a", fs.extension("file.a"));
+	EXPECT_EQ("", fs.extension("file."));
+	EXPECT_EQ("file", fs.extension(".file"));
```

Comment:
You aren't testing for paths that do not contain a dot or have no extension.

The extension function should really include the dot in the return value to distinguish between files that end in a dot (`"file."` returns `"."`) and files that do not have an extension at all (`"file"` returns `""`.)

Reply:
Including the "." as part of the extension is exactly the kind of thing we need to discuss about what we want going forward.

Good point about testing a path with no ".".

I skipped over testing of directory names with an extension and a trailing "/", because they don't work. As in, the following would fail:
```cpp
EXPECT_EQ("ext", fs.extension("DirName.ext/"));  // Actually returns "ext/"
```

Reply:
> 
> I skipped over testing of directory names with an extension and a trailing "/", because they don't work. As in, the following would fail:
> 
> ```c++
> EXPECT_EQ("ext", fs.extension("DirName.ext/"));  // Actually returns "ext/"
> ```

This goes back to my suggestion of parroting the standard. The first thing the standard does is check against the special paths `dot` and `dot dot` (`.`, `..`). If the paths are equal to either of those two, it returns and empty string. Otherwise is calls `filename()` which has its own rules to extract the filename, including if there is a trailing slash, and that is what `extension()` operates on. `filename()` may very well return an empty string and if it does, `extension()` is done and it just returns that.